### PR TITLE
simplify model saving code and NonNull

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
I hope this will get a PR simple to merge.

I was dealing with savings part of the code. I realized that this porting of python code could be improved (at least according to my taste) by indicating that if a model is given to `save` it should be nonNull. Then `save(model)` would called `save()` instead of the reverse.

I followed all call to `save` to esure that indeed, `save(model)` is never called with a null value.

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Models.java [hunk 1]
```diff
@@ -178,18 +180,18 @@ public void save(JSONObject m) {
...
-			m.put("usn", mCol.usn());
-			// TODO: fix empty id problem on _updaterequired (needed for model adding)
-			if (!isModelNew(m)) {
-				_updateRequired(m);
-			}
-			if (templates) {
-				_syncTemplates(m);
-			}
+        if (m.has("id")) {
+            m.put("mod", Utils.intTime());
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Models.java:184]: whitespace change - could you split into another PR for ease of reviewing

File: AnkiDroid/src/main/java/com/ichi2/libanki/Models.java [hunk 2]
```diff
@@ -178,18 +180,18 @@ public void save(JSONObject m) {
...
+            // TODO: fix empty id problem on _updaterequired (needed for model adding)
+            if (!isModelNew(m)) {
+                _updateRequired(m);
+            }
+            if (templates) {
+                _syncTemplates(m);
+            }
         }
-        mChanged = true;
+        save();
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Models.java:194]: This is unintuitive - in Java, methods typically become more specific, this makes me double-take

File: AnkiDroid/src/main/java/com/ichi2/libanki/Models.java (hunk 1 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Models.java:184]: https://github.com/ankidroid/Anki-Android/pull/6359

File: AnkiDroid/src/main/java/com/ichi2/libanki/Models.java (hunk 2 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Models.java:194]: Oh okay. I don't do enough java I get. Seems far more intuitive to me to ask that if model is given, it is not null, then we do specific change to models, and then, another method do the part which don't care about model

**Comment by Arthur-Milchior** [comment]
> Does this add anything in terms of performance?
No. I would have added "NF" if I know the abbreviation when I made this PR. I just found it far more intuitive to state that the method with a model except to actually have a model and that the code which is executed in any case belongs to a method without a model variable.

If it's not proper java, then I can remove this part. But I find it really more complex to read right now. I mean, `save()` states "Eh, I'm going to send null to `save(model)` and this other method will need to have extra code to deal with it makes no sens to me.

**Comment by david-allison-1** [comment]
I think it's the same in most programming languages (likely why the Python was structured the way it is) - when overloading aim to call from fewer -> greater parameters. I've had a Google and the main reference I found was: https://stackoverflow.com/a/50942285

**Comment by Arthur-Milchior** [comment]
I'm not really use of programming language with overloading.
So I guess we close this PR then, right ?

**Comment by david-allison-1** [comment]
Might as well close, whitespace change has gone in in #6359
